src/chainstate/stacks/boot/pox.clar

```diff
@@ -380,6 +461,140 @@
       (ok { stacker: tx-sender, lock-amount: amount-ustx, unlock-burn-height: (reward-cycle-to-burn-height (+ first-reward-cycle lock-period)) }))
 )
 
+(define-public (revoke-delegate-stx (delegator principal))
+  (begin
+    ;; must be called directly by the tx-sender or by an allowed contract-caller
+    (asserts! (check-caller-allowed)
+              (err ERR_STACKING_PERMISSION_DENIED))
+    (ok (map-delete delegation-state { stacker: tx-sender }))))
+
+;; Delegate to `delegator` the ability to stack from a given address.
+;;  This method _does not_ lock the funds, rather, it allows the delegator
+;;  to issue the stacking lock.
+;; The caller specifies:
+;;   * amount-ustx: the total amount of ustx the delegator may be allowed to lock
+;;   * until-burn-ht: an optional burn height at which this delegation expiration
+;;   * pox-addr: an optional address to which any rewards *must* be sent
+(define-public (delegate-stx (amount-ustx uint)
+                             (delegator principal)
+                             (until-burn-ht (optional uint))
+                             (pox-addr (optional { version: (buff 1),
+                                                   hashbytes: (buff 20) })))
+    (begin
+      ;; must be called directly by the tx-sender or by an allowed contract-caller
+      (asserts! (check-caller-allowed)
+                (err ERR_STACKING_PERMISSION_DENIED))
+
+      ;; tx-sender principal must not be stacking
+      (asserts! (is-none (get-stacker-info tx-sender))
+        (err ERR_STACKING_ALREADY_STACKED))
+
+      ;; tx-sender must not be delegating
+      (asserts! (is-none (get-check-delegation tx-sender))
+        (err ERR_STACKING_ALREADY_DELEGATED))
+
+      ;; add delegation record
+      (map-set delegation-state
+        { stacker: tx-sender }
+        { amount-ustx: amount-ustx,
+          delegated-to: delegator,
+          until-burn-ht: until-burn-ht,
+          pox-addr: pox-addr })
+
+      (ok true)))
+
+;; Commit partially stacked STX.
+;;   this allows a stacker/delegator to lock fewer STX than the minimal threshold in multiple transactions,
```

Comment:
Can you add punctuation and capitalization to this?  The code comments here will be set in stone forever once the system launches, so they should be "documentation-quality" ;)

Reply:
Added in c833b65ec